modules/data.remote/R/call_MODIS.R

```diff
@@ -27,65 +29,116 @@
 ##' 
 ##' @author Bailey Morrison
 ##'  
-call_MODIS <- function(outfolder = ".", start_date, end_date, lat, lon, size = 0, product, band, band_qc = "", band_sd = "", package_method = "MODISTools") {
+call_MODIS <- function(outfolder = NULL, start_date, end_date, lat, lon, size = 0, product, band, band_qc = "", band_sd = "", siteID = NULL, package_method = "MODISTools", QC_filter = F) {
   
   # makes the query search for 1 pixel and not for rasters for now. Will be changed when we provide raster output support.
   size <- 0
+# reformat start and end date if they are in YYYY/MM/DD format instead of YYYYJJJ
+  if (grepl("/", start_date) == T)
+  {
+    start_date = as.Date(paste0(lubridate::year(start_date), spatial.tools::add_leading_zeroes(lubridate::yday(start_date), 3)), format = "%Y%j")
+  }
   
-  # set start and end dates to correct format
-  if (package_method == "MODISTools"){
-    
+  if (grepl("/", end_date) == T)
+  {
+    end_date = as.Date(paste0(lubridate::year(end_date), spatial.tools::add_leading_zeroes(lubridate::yday(end_date), 3)), format = "%Y%j")
+  }
+  start_date = as.Date(start_date, format = "%Y%j")
+  end_date = as.Date(end_date, format = "%Y%j")
```

Comment:
I'm Mr. Annoying Style Cop, but please follow our style guide and use `<-` for assignment and reserve `=` for function arguments. Similarly, please use `TRUE/FALSE` instead of `T/F`. The former are reserved words in R, while the latter are not:

 ``` r
# These will error 
TRUE <- 0
#> Error in TRUE <- 0: invalid (do_set) left-hand side to assignment
FALSE <- 5
#> Error in FALSE <- 5: invalid (do_set) left-hand side to assignment
# ...but these will not!
T <- FALSE
F <- 3
if (T) {
  print("yes")
} else {
  print("no")
}
#> [1] "no"
if (F) print("This is true!")
#> [1] "This is true!"
```

Reply:
Yup, @bailsofhay what I also keep telling you ;)